### PR TITLE
ARTEMIS-1619 - Add plugin support for address lifecyle

### PR DESCRIPTION
**Pull request by cshannon**
Adding callbacks to the plugin API for address creation, update and
removals

### CODE REVIEW

File: tests/integration-tests/src/test/java/org/apache/activemq/artemis/tests/integration/plugin/StompPluginTest.java [hunk 1]
```diff
@@ -127,10 +131,33 @@ public void testSendAndReceive() throws Exception {
...
+         // unsub
+         unsubscribe(newConn, "a-sub");
+         newConn.disconnect();
+
+         verifier.validatePluginMethodsAtLeast(1, BEFORE_ADD_ADDRESS, AFTER_ADD_ADDRESS,
+               BEFORE_REMOVE_ADDRESS, AFTER_REMOVE_ADDRESS);
 
       } catch (Throwable e) {
-         e.printStackTrace();
+         fail(e.getMessage());
```
Comment (clebertsuconic) [inline tests/integration-tests/src/test/java/org/apache/activemq/artemis/tests/integration/plugin/StompPluginTest.java]: the printStackTrace is useful when it fails :)
I have been there.

Comment (clebertsuconic) [reply tests/integration-tests/src/test/java/org/apache/activemq/artemis/tests/integration/plugin/StompPluginTest.java]: this is actually the only request I'm going to make.. almost merged this.

Comment (clebertsuconic) [reply tests/integration-tests/src/test/java/org/apache/activemq/artemis/tests/integration/plugin/StompPluginTest.java]: hmmmm.. but it was being ignored before?

maybe just remove the catch and let it fail? remove the throws Exception on the test?

Comment (cshannon) [reply tests/integration-tests/src/test/java/org/apache/activemq/artemis/tests/integration/plugin/StompPluginTest.java]: yeah i can remove it, i'm not sure why it's there at all

Comment (cshannon) [reply tests/integration-tests/src/test/java/org/apache/activemq/artemis/tests/integration/plugin/StompPluginTest.java]: ok i updated the PR

**Pull request closed without merging**